// @neynar/nodejs-sdk/build/neynar-api/v2/openapi-farcaster/apis/fname-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Farcaster API V2
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FnameApi = exports.FnameApiFactory = exports.FnameApiFp = exports.FnameApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * FnameApi - axios parameter creator
 * @export
 */
const FnameApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Check if a given fname is available
         * @summary Check fname availability
         * @param {string} fname
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<FnameAvailabilityResponse>} A promise that resolves to a `FnameAvailabilityResponse` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/is-fname-available)
         *
         */
        isFnameAvailable: async (fname, options = {}) => {
            // verify required parameter 'fname' is not null or undefined
            (0, common_1.assertParamExists)('isFnameAvailable', 'fname', fname);
            const localVarPath = `/farcaster/fname/availability`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            if (fname !== undefined) {
                localVarQueryParameter['fname'] = fname;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.FnameApiAxiosParamCreator = FnameApiAxiosParamCreator;
/**
 * FnameApi - functional programming interface
 * @export
 */
const FnameApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.FnameApiAxiosParamCreator)(configuration);
    return {
        /**
         * Check if a given fname is available
         * @summary Check fname availability
         * @param {string} fname
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<FnameAvailabilityResponse>} A promise that resolves to a `FnameAvailabilityResponse` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/is-fname-available)
         *
         */
        async isFnameAvailable(fname, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.isFnameAvailable(fname, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['FnameApi.isFnameAvailable']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.FnameApiFp = FnameApiFp;
/**
 * FnameApi - factory interface
 * @export
 */
const FnameApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.FnameApiFp)(configuration);
    return {
        /**
         * Check if a given fname is available
         * @summary Check fname availability
         * @param {FnameApiIsFnameAvailableRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<FnameAvailabilityResponse>} A promise that resolves to a `FnameAvailabilityResponse` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/is-fname-available)
         *
         */
        isFnameAvailable(requestParameters, options) {
            return localVarFp.isFnameAvailable(requestParameters.fname, options).then((request) => request(axios, basePath));
        },
    };
};
exports.FnameApiFactory = FnameApiFactory;
/**
 * FnameApi - object-oriented interface
 * @export
 * @class FnameApi
 * @extends {BaseAPI}
 */
class FnameApi extends base_1.BaseAPI {
    /**
     * Check if a given fname is available
     * @summary Check fname availability
     * @param {FnameApiIsFnameAvailableRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof FnameApi
     * @returns {Promise<FnameAvailabilityResponse>} A promise that resolves to a `FnameAvailabilityResponse` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/is-fname-available)
     *
     */
    isFnameAvailable(requestParameters, options) {
        return (0, exports.FnameApiFp)(this.configuration).isFnameAvailable(requestParameters.fname, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.FnameApi = FnameApi;
