// @neynar/nodejs-sdk/build/neynar-api/v2/openapi-farcaster/models/embed-type.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Farcaster API V2
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmbedType = void 0;
/**
 * The content type of the embedded file
 * @export
 * @enum {string}
 */
exports.EmbedType = {
    Text: 'text',
    Image: 'image',
    Video: 'video',
    Audio: 'audio',
    TextHtml: 'text/html',
    TextPlain: 'text/plain',
    ImageJpeg: 'image/jpeg',
    ImagePng: 'image/png',
    ImageGif: 'image/gif',
    ImageWebp: 'image/webp',
    ImageSvgxml: 'image/svg+xml',
    ImageHeif: 'image/heif',
    VideoMp4: 'video/mp4',
    VideoQuicktime: 'video/quicktime',
    AudioMpeg: 'audio/mpeg',
    ApplicationPdf: 'application/pdf',
    ApplicationJson: 'application/json',
    ApplicationXMpegurl: 'application/x-mpegurl'
};
