// @neynar/nodejs-sdk/build/neynar-api/common/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotificationType = exports.ValidateFrameAggregateWindow = exports.BulkUserAddressTypes = exports.BulkCastsSortType = exports.TrendingFeedTimeWindow = exports.TimeWindow = exports.SIGNED_KEY_REQUEST_TYPE_FOR_ADD_FOR = exports.SIGNED_KEY_REQUEST_VALIDATOR = exports.SIGNED_KEY_REQUEST_TYPE = exports.SIGNED_KEY_REQUEST_VALIDATOR_EIP_712_DOMAIN = void 0;
// DO NOT CHANGE ANY VALUES IN THIS CONSTANT
exports.SIGNED_KEY_REQUEST_VALIDATOR_EIP_712_DOMAIN = {
    name: "Farcaster SignedKeyRequestValidator",
    version: "1",
    chainId: 10,
    verifyingContract: "0x00000000fc700472606ed4fa22623acf62c60553",
};
// DO NOT CHANGE ANY VALUES IN THIS CONSTANT
exports.SIGNED_KEY_REQUEST_TYPE = [
    { name: "requestFid", type: "uint256" },
    { name: "key", type: "bytes" },
    { name: "deadline", type: "uint256" },
];
exports.SIGNED_KEY_REQUEST_VALIDATOR = {
    name: "Farcaster SignedKeyRequestValidator",
    version: "1",
    chainId: 10,
    verifyingContract: "0x00000000fc700472606ed4fa22623acf62c60553",
};
exports.SIGNED_KEY_REQUEST_TYPE_FOR_ADD_FOR = [
    { name: "owner", type: "address" },
    { name: "keyType", type: "uint32" },
    { name: "key", type: "bytes" },
    { name: "metadataType", type: "uint8" },
    { name: "metadata", type: "bytes" },
    { name: "nonce", type: "uint256" },
    { name: "deadline", type: "uint256" },
];
var TimeWindow;
(function (TimeWindow) {
    TimeWindow["ONE_DAY"] = "1d";
    TimeWindow["SEVEN_DAYS"] = "7d";
    TimeWindow["THIRTY_DAYS"] = "30d";
})(TimeWindow || (exports.TimeWindow = TimeWindow = {}));
var TrendingFeedTimeWindow;
(function (TrendingFeedTimeWindow) {
    TrendingFeedTimeWindow["ONE_HOUR"] = "1h";
    TrendingFeedTimeWindow["SIX_HOUR"] = "6h";
    TrendingFeedTimeWindow["TWELVE_HOUR"] = "12h";
    TrendingFeedTimeWindow["TWENTY_FOUR_HOUR"] = "24h";
})(TrendingFeedTimeWindow || (exports.TrendingFeedTimeWindow = TrendingFeedTimeWindow = {}));
var BulkCastsSortType;
(function (BulkCastsSortType) {
    BulkCastsSortType["TRENDING"] = "trending";
    BulkCastsSortType["LIKES"] = "likes";
    BulkCastsSortType["RECASTS"] = "recasts";
    BulkCastsSortType["REPLIES"] = "replies";
    BulkCastsSortType["RECENT"] = "recent";
})(BulkCastsSortType || (exports.BulkCastsSortType = BulkCastsSortType = {}));
var BulkUserAddressTypes;
(function (BulkUserAddressTypes) {
    BulkUserAddressTypes["CUSTODY_ADDRESS"] = "custody_address";
    BulkUserAddressTypes["VERIFIED_ADDRESS"] = "verified_address";
})(BulkUserAddressTypes || (exports.BulkUserAddressTypes = BulkUserAddressTypes = {}));
var ValidateFrameAggregateWindow;
(function (ValidateFrameAggregateWindow) {
    ValidateFrameAggregateWindow["TEN_SECONDS"] = "10s";
    ValidateFrameAggregateWindow["ONE_MINUTE"] = "1m";
    ValidateFrameAggregateWindow["TWO_MINUTES"] = "2m";
    ValidateFrameAggregateWindow["FIVE_MINUTES"] = "5m";
    ValidateFrameAggregateWindow["TEN_MINUTES"] = "10m";
    ValidateFrameAggregateWindow["TWENTY_MINUTES"] = "20m";
    ValidateFrameAggregateWindow["THIRTY_MINUTES"] = "30m";
    ValidateFrameAggregateWindow["TWO_HOURS"] = "2h";
    ValidateFrameAggregateWindow["TWELVE_HOURS"] = "12h";
    ValidateFrameAggregateWindow["ONE_DAY"] = "1d";
    ValidateFrameAggregateWindow["SEVEN_DAYS"] = "7d";
})(ValidateFrameAggregateWindow || (exports.ValidateFrameAggregateWindow = ValidateFrameAggregateWindow = {}));
var NotificationType;
(function (NotificationType) {
    NotificationType["LIKES"] = "likes";
    NotificationType["RECASTS"] = "recasts";
    NotificationType["REPLIES"] = "replies";
    NotificationType["FOLLOWS"] = "follows";
    NotificationType["MENTIONS"] = "mentions";
})(NotificationType || (exports.NotificationType = NotificationType = {}));
